lesson14.html

```diff
@@ -0,0 +1,172 @@
+        width: 120px;
+        padding: 5px 10px;
+        left: 0;
+        transform: translate(-15%, -140%);
+      }
+      .tooltip1:hover .tooltiptext1 {
+        visibility: visible;
+        opacity: 1;
+      }
+      .tooltip1 .tooltiptext1::after {
+        content: " ";
+        position: absolute;
+        top: 100%;
+        left: 50%;
+        border-left: 10px solid transparent;
+        border-right: 10px solid transparent;
+        border-top: 10px solid gray;
+        transform: translateX(-50%);
+      }
+      .tooltip2 .tooltiptext2 {
+        top: -50%;
+        width: 120px;
+        padding: 9px;
+        left: 0;
+        transform: translateX(77%);
+      }
+      .tooltip2:hover .tooltiptext2 {
+        visibility: visible;
+        opacity: 1;
+      }
+      .tooltip2 .tooltiptext2::after {
+        content: " ";
+        position: absolute;
+        top: 50%;
+        left: -10px;
+        border-bottom: 10px solid transparent;
+        border-right: 10px solid gray;
+        border-top: 10px solid transparent;
+        transform: translateY(-50%);
+      }
+      .tooltip3 .tooltiptext3 {
+        top: 110%;
+        width: 110px;
+        padding: 10px 20px;
+        left: 50%;
+        transform: translateX(-50%);
+      }
+      .tooltip3:hover .tooltiptext3 {
+        visibility: visible;
+        opacity: 1;
+      }
+      .tooltip3 .tooltiptext3::after {
+        content: " ";
+        position: absolute;
+        top: -10%;
+        left: 80px;
+        border-left: 10px solid transparent;
+        border-right: 10px solid transparent;
+        border-bottom: 10px solid gray;
+        transform: translateX(-50%);
+      }
+      .tooltip4 .tooltiptext4 {
+        transform: translateY(-50%);
+        top: 50%;
+        right: 109%;
+        width: 120px;
+        padding: 9px;
+      }
+      .tooltip4:hover .tooltiptext4 {
+        visibility: visible;
+        opacity: 1;
+      }
+      .tooltip4 .tooltiptext4::after {
+        content: " ";
+        position: absolute;
+        top: 50%;
+        left: 100%;
+        transform: translateY(-50%);
+        border-top: 10px solid transparent;
+        border-bottom: 10px solid transparent;
+        border-left: 10px solid gray;
+      }
+    </style>
+  </head>
+  <body>
+    <p>
+      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Et, perferendis!
+      Repudiandae deleniti, nostrum blanditiis impedit laudantium, numquam atque
+      itaque illo facere odio ut tempore quibusdam obcaecati, recusandae minus
+      corporis rem!
+    </p>
+    <p>
+      Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam voluptatum
+      nemo quo ullam deleniti rem maiores a dolor maxime praesentium minus earum
+      optio sapiente voluptatibus facere, blanditiis ea tempore facilis!
+    </p>
+    <div class="container_blue">
+      <div class="trigger tooltip1">
+        Trigger top
+        <span class="tooltiptext1 tttgeneral">
```

Comment:
I suppose you can keep just 1 class here. for example ```tooltiptext```
and use it as general. You anyway style all the tooltiptext based on their parents. So there is no need to create unique classnames